bff-service/index.js

```diff
@@ -0,0 +1,65 @@
+require('dotenv').config();
+const express = require('express');
+const axios = require('axios').default;
+
+const app = express();
+const PORT = process.env.PORT || 3001;
+
+const cachedData = {};
+
+app.use(express.json());
+
+app.all('/*', (req, res) => {
+    const { originalUrl, method, body } = req;
+
+    const recipient = originalUrl.split('/')[1];
+    const recipientUrl = process.env[recipient];
+    const requestUrl = `${recipientUrl}${originalUrl}`;
+
+    if (recipientUrl) {
+        const isCachedUrl =
+            method === 'GET' && originalUrl.endsWith(process.env.cached_url);
+
+        const config = {
+            method,
+            url: requestUrl,
+            ...(Object.keys(req.body || {}).length > 0 && { data: body }),
+        };
+
+        if (
+            isCachedUrl &&
+            cachedData[requestUrl] &&
+            cachedData[requestUrl].expiredAt > Date.now()
```

Comment:
as alternative, you can use node-cache lib, it provides some api to manage keys and automatically expires the data